File: Examen/Examen_1er_Bimestre/src/com/ejemplo/controller/LibroController.java
package java.com.ejemplo.controller;

import java.com.ejemplo.model.Libro;
import java.com.ejemplo.service.LibroService;

import java.io.IOException;
import java.util.List;

public class LibroController {
    private LibroService libroService = new LibroService();

    // Crear un nuevo libro
    public void createLibro(Libro libro) throws IOException {
        libroService.saveLibro(libro);
    }

    // Obtener todos los libros
    public List<Libro> getLibros() throws IOException {
        return libroService.getAllLibros();
    }

    // Actualizar un libro existente
    public void updateLibro(Libro libro) throws IOException {
        libroService.updateLibro(libro);
    }

    // Eliminar un libro por ID
    public void deleteLibro(int id) throws IOException {
        libroService.deleteLibro(id);
    }

    // Obtener un libro por ID
    public Libro getLibroById(int id) throws IOException {
        return libroService.getLibroById(id);
    }
}

File: Examen_02/src/controller/LibroController.kt
package controller

import dao.LibroDao
import model.Libro
import java.io.IOException


class LibroController {
    private val libroDao = LibroDao()

    @Throws(IOException::class)
    fun createLibro(libro: Libro) {
        libroDao.save(libro)
    }

    @Throws(IOException::class)
    fun getLibros(): List<Libro> {
        return libroDao.findAll()
    }

    @Throws(IOException::class)
    fun updateLibro(libro: Libro) {
        libroDao.update(libro)
    }

    @Throws(IOException::class)
    fun deleteLibro(id: Int) {
        libroDao.delete(id)
    }

    @Throws(IOException::class)
    fun getLibroById(id: Int): Libro? {
        return libroDao.findById(id)
    }

    @Throws(IOException::class)
    fun getLibrosByAutor(autorId: Int): List<Libro> {
        return libroDao.findByAutorId(autorId)
    }
}
